lib/ami_spec/aws_instance.rb

```diff
@@ -61,7 +62,7 @@ def instances_options
         key_name: @key_name,
         network_interfaces: [{
           device_index: 0,
-          associate_public_ip_address: @public_ip,
+          associate_public_ip_address: @public_ip || @associate_public_ip,
```

Comment:
Won't there always be a public ip?

Reply:
What about `ssh_ip_address`?

Reply:
> Won't there always be a public ip?

The default behaviour of AmiSpec is not to associate a public IP address.
In retrospect that was a mistake and we probably should've left it empty so the subnet defaults decide.